src/index.js

```diff
@@ -1,59 +1,33 @@
 import React from 'react';
 import PropTypes from 'prop-types';
-import Payment from 'payment';
+import creditCardType from 'credit-card-type';
+import luhn from 'luhn';
+
+import { configure } from './config';
+import { sanitizeNumber } from './utils';
 
 class ReactCreditCards extends React.Component {
   constructor(props) {
     super(props);
 
-    this.setCards();
+    const validCardTypes = configure();
+
+    this.state = {
+      validCardTypes,
+    };
   }
 
-  static propTypes = {
-    acceptedCards: PropTypes.array,
-    callback: PropTypes.func,
-    cvc: PropTypes.oneOfType([
-      PropTypes.string,
-      PropTypes.number,
-    ]).isRequired,
-    expiry: PropTypes.oneOfType([
-      PropTypes.string,
-      PropTypes.number,
-    ]).isRequired,
-    focused: PropTypes.string,
-    issuer: PropTypes.string,
-    locale: PropTypes.shape({
-      valid: PropTypes.string,
-    }),
-    name: PropTypes.string.isRequired,
-    number: PropTypes.oneOfType([
-      PropTypes.string,
-      PropTypes.number,
-    ]).isRequired,
-    placeholders: PropTypes.shape({
-      name: PropTypes.string,
-    }),
-    preview: PropTypes.bool,
-  };
-
-  static defaultProps = {
-    acceptedCards: [],
-    locale: {
-      valid: 'valid thru',
-    },
-    placeholders: {
-      name: 'YOUR NAME HERE',
-    },
-    preview: false,
-  };
+  componentDidMount() {
```

Comment:
This would break SSR since the component won't mount until it's rendered in a browser.

Reply:
good catch, moved back to the constructor!